### DISCUSSION TITLE
Configuration.ShouldQuote - return true for only fields on the DTO which are strings

### DISCUSSION THREAD

**Discussion by adrianwright109** [JoshClose/CsvHelper#1329] [Q&A]
Hi have a DTO object with a number of properties with different types, `string`, `int` or `bool` etc.

I would like to apply double quotes around the `string` properties only.

In the `Configuration.ShouldQuote` the `field` parameter values have all been converted to `string` so there is no way of knowing whether the original type from the DTO was a `string`, `int` or `bool`.

Is there a way I can find the underlying property type from the DTO so that I can pass back `true` from `Configuration.ShouldQuote` only for fields that were originally of type `string`?

```
public class TestDTO
{
    public string Field1 { get; set; }
    public int Field2 { get; set; }
    public bool Field3 { get; set; }
}
```

```
    var rawData = new[]
    {
        new TestDTO { Field1 = "Field1", Field2 = 1, Field3 = true },
        new TestDTO { Field1 = "Field2", Field2 = 10, Field3 = false }
    };
    using (var writer = new StreamWriter("file.csv"))
    {
        using (var csv = new CsvWriter(writer))
        {
            csv.Configuration.ShouldQuote = (field, context) =>
            {
                return field is string; // doesn't work as all fields at this point are strings
            };
            
            csv.WriteRecords(rawData);
        }
    }
```

**Comment by AltruCoder** [comment]
Answered on [StackOverflow](https://stackoverflow.com/questions/56579848/csvhelper-configuration-shouldquote-return-true-for-only-fields-on-the-dto-whi)
